lib/pbench/server/api/resources/metadata_api.py

```diff
@@ -0,0 +1,445 @@
+        else:
+            response_object = {
+                "data": metadata_object.get_json(["id", "created", "updated", "key"]),
+            }
+            return make_response(jsonify(response_object), 200)
+
+    @Auth.token_auth.login_required()
+    def delete(self, id):
+        """
+        Delete request for deleting a metadata object of a user given a metadata id.
+        This requires a Pbench auth token in the header field.
+        Public metadata objects can only be deleted by admin users.
+
+        The url requires a metadata object id such as /user/metadata/<int:id>
+
+        Required headers include
+            Authorization:   Bearer <Pbench authentication token (user received upon login)>
+
+        :return: JSON Payload
+            response_object = {
+                    "message": "success"
+                }
+        """
+        if id is None:
+            self.logger.warning("Metadata id not provided during metadata query")
+            abort(400, message="Missing metadata id in the URI for delete operation")
+
+        try:
+            # Fetch the metadata object
+            metadata_objects = UserMetadata.query(id=id)
+        except Exception:
+            self.logger.exception(
+                "Exception occurred in the Delete request while querying the Metadata model, id: {}",
+                id,
+            )
+            abort(500, message="INTERNAL ERROR")
+
+        if metadata_objects:
+            metadata_object = metadata_objects[0]
+        else:
+            abort(404, message="Not found")
+
+        # Verify if the metadata object id in the url belongs to the logged in user
+        if not verify_metadata(metadata_object, self.logger):
+            abort(403, message="Not authorized to DELETE the metadata object")
+
+        try:
+            # Delete the metadata object
+            UserMetadata.delete(id=id)
+        except Exception:
+            self.logger.exception(
+                "Exception occurred in the while deleting the metadata entry, id: {}",
+                id,
+            )
+            abort(500, message="INTERNAL ERROR")
+
+        response_object = {
+            "message": "Metadata object deleted",
+        }
+        return make_response(jsonify(response_object), 200)
+
+
+class PublishMetadata(Resource):
+    """
+    Abstracted pbench API for handling user metadata
+    """
+
+    def __init__(self, config, logger):
+        self.server_config = config
+        self.logger = logger
+
+    @Auth.token_auth.login_required()
+    def post(self, id):
+        """
+        Post request for publishing the metadata object for public access.
+        This requires a Pbench auth token in the header field. Only authorized users can make their metadata public.
+        Right now once the metadata object is made public it can not be reverted back to the private entity
+        TODO: Is it desirable to be able to revert the object back as a private entity
```

Comment:
Yeah we didn't discuss about it, this was a prototype idea that I thought would be good to have. As you correctly pointed out it enables user to publish a certain view if they want to and that view would be accessible to all the logged-in and non logged-in users. If users are comfortable with publishing their dataset, they may want to publish their metadata view as well for others to access (and even for themselves without logging in next time). 

I am not sure if this would add much of a value for the user experience, but an option we could give. However, I am certainly open for any suggestions.